src/EditorFeatures/Core/Extensibility/NavigationBar/NavigationBarProjectItem.cs

```diff
@@ -2,29 +2,27 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
+using System;
+using System.Collections.Immutable;
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor
 {
-    internal sealed class NavigationBarProjectItem : NavigationBarItem
+    internal sealed class NavigationBarProjectItem : NavigationBarItem, IEquatable<NavigationBarProjectItem>
```

Comment:
implemented equality to easily check if the new project items we've computed equal the last ones we presented.